Client/Application/AppDelegate.swift

```diff
@@ -218,43 +218,50 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     }
 
     func application(application: UIApplication, openURL url: NSURL, sourceApplication: String?, annotation: AnyObject) -> Bool {
-        if let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) {
-            if components.scheme != "firefox" && components.scheme != "firefox-x-callback" {
-                return false
-            }
-            var url: String?
-            var isPrivate: Bool = false
-            for item in (components.queryItems ?? []) as [NSURLQueryItem] {
-                switch item.name {
-                case "url":
-                    url = item.value
-                case "private":
-                    isPrivate = (item.value != "false" && item.value != "0")
-                default: ()
-                }
+        guard let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) else {
+            return false
+        }
+        if components.scheme != "firefox" && components.scheme != "firefox-x-callback" {
+            return false
+        }
+        var url: String?
+        var isPrivate: Bool = false
+        for item in (components.queryItems ?? []) as [NSURLQueryItem] {
+            switch item.name {
+            case "url":
+                url = item.value
+            case "private":
+                isPrivate = (item.value != "false" && item.value != "0")
```

Comment:
this is where I miss NSString which has a boolValue method. I guess you could convert to NSString and then compare there. I couldn't test it as I couldn't get this to build under Xcode 7.3 (stupid compiler meltdown, nothing to do with your code)

`isPrivate = NSString(item.value).boolValue()`